client/app/lib/onboarding.coffee

```diff
@@ -0,0 +1,111 @@
+# Local class Step
+class Step
+    # Retrieves properties from config Step plain object
+    constructor: (step) ->
+        if step
+            ['name', 'route', 'view'].forEach (property) =>
+                @[property] = step[property]
+
+
+    # Record handlers for 'validated' internal pseudo-event
+    onValidated: (callback) ->
+        throw new Error 'Callback parameter should be a function' \
+            unless typeof callback is 'function'
+        @validatedHandlers = @validatedHandlers or []
+        @validatedHandlers.push callback
+
+
+    # Trigger 'validated' pseudo-event
+    triggerValidated: () ->
+        if @validatedHandlers
```

Comment:
@gregorylegarec : I find a contradiction between this line and line 14 above. Furthermore I don't find any default value such as : 
`````
class Step

    validatedHandlers: null
````